pennylane/tape/tape.py

```diff
@@ -237,6 +237,47 @@ def stop_at(obj):  # pylint: disable=unused-argument
     return new_tape
 
 
+def expand_tape_state_prep(tape, skip_first=True):
+    """Expand all instances of StatePrep operations in the tape.
+
+    Args:
+        tape (QuantumTape): The tape to expand.
+        skip_first (Bool): If ``True``, will not expand a StatePrep operation if
+            it is the first operation in the tape.
+
+    Returns:
+        QuantumTape: The expanded version of ``tape``.
+
+    **Example**
+    ...
+    """
+    new_prep = []
+    new_ops = []
+
+    for i, op in enumerate(tape._prep + tape._ops):
+        if isinstance(op, StatePrep):
```

Comment:
It seems to cover `qml.BasisState` as well. Not sure if it creates any problems though.

Reply:
Nope that is by design, both are "StatePrep" operations, these PRs will let any operator which inherits from StatePrep be used mid-circuit